src/main/java/com/github/distributedtracing/ContinuationLocal.java

```diff
@@ -0,0 +1,170 @@
+package com.github.distributedtracing;
+
+public final class ContinuationLocal<T> {
+    private static InheritableThreadLocal<Object[]> localCtx =
```

Comment:
I'm just questioning whether implicit passing via InheritableThreadLocal is a fools errand for the reasons you mention. I consider wrapping Executors to be explicit passing btw and something most modern application do as a matter of routine.